examples/instance_kind/README.md

```diff
@@ -0,0 +1,190 @@
+-->
+
+# Model Instance Kind Example
+
+Triton allows an execution of a model to be performed on both CPU, and GPU. 
+In Python Backend this can be achieved by specifying an 
+[`instance_group`](https://github.com/triton-inference-server/server/blob/main/docs/user_guide/model_configuration.md#instance-groups) 
+parameter in a config file.
+
+In this section, we demonstrate how to specify a `instance kind` for your python model.
+We will use a `ResNet50` model as our base model for this example.
+
+## Create a ResNet50 model repository
+
+We will use the files that come with this example to create the model
+repository.
+
+First, download the [client.py](client.py), [config.pbtxt](config.pbtxt),
+[resnet50_labels.txt](resnet50_labels.txt), and [model.py](model.py) 
+to your local machine.
+
+Next, in the same directory with the four aformentioned files, create the model
+repository with the following commands:
+```
+mkdir -p models/resnet50/1 &&
+mv model.py models/resnet50/1/ &&
+mv config.pbtxt models/resnet50/
+```
+
+## Pull the Triton Docker images
+
+We need to install Docker and NVIDIA Container Toolkit before proceeding, refer
+to the
+[installation steps](https://github.com/triton-inference-server/server/tree/main/docs#installation).
+
+To pull the latest containers, run the following commands:
+```
+$ docker pull nvcr.io/nvidia/tritonserver:<yy.mm>-py3
+$ docker pull nvcr.io/nvidia/tritonserver:<yy.mm>-py3-sdk
+```
+See the installation steps above for the `<yy.mm>` version.
+
+For example, if the latest version is `23.01`, the above commands translate to the
+following:
+```
+$ docker pull nvcr.io/nvidia/tritonserver:23.01-py3
+$ docker pull nvcr.io/nvidia/tritonserver:23.01-py3-sdk
+```
+
+Be sure to replace the `<yy.mm>` with the version pulled for all the remaining
+parts of this example.
+
+## Start the Triton Server
+
+At the directory where we copied our resnet50 model (at where the "models" folder
+is located), run the following command:
+```
+$ docker run --gpus all --shm-size 1G -it --rm -p 8000:8000 -v `pwd`:/instance_kind nvcr.io/nvidia/tritonserver:<yy.mm>-py3 /bin/bash
+```
+
+Inside the container, we need to install `torch` and `pillow` to run this example. 
+We recommend to use `pip` method for the installation:
+
+```
+pip3 install torch==1.13.0+cu117 -f https://download.pytorch.org/whl/torch_stable.html pillow
+```
+
+Finally, we need to start the Triton Server:
+```
+tritonserver --model-repository /instance_kind/models
+```
+
+To leave the container for the next step, press: `CTRL + P + Q`.
+
+## Start the Triton SDK Container and Test Inference
+
+To start the sdk container, run the following command:
+```
+docker run --gpus all --network=host --pid=host --ipc=host -v `pwd`:/instance_kind -ti nvcr.io/nvidia/tritonserver:<yy.mm>-py3-sdk /bin/bash
+```
+
+The `client.py` requires the following packages to be installed: `torch`, `torchvision`, `pillow` and `validators`. 
```

Comment:
I was wondering where is `validators` used?

Reply:
Client uses utils, loaded from `torch.hub` to preprocess image from url. The exact `utils` is described here with all imports, including `validators` 

https://github.com/NVIDIA/DeepLearningExamples/blob/a8c42a142ceb67513892a303b48185d9c52864c0/PyTorch/Classification/ConvNets/image_classification/models/entrypoints.py#L43